File: wallet-android/chapter5/code/app/src/main/java/com/digquant/util/ResourceUtil.java
package com.digquant.util;

import android.annotation.SuppressLint;
import android.content.Context;
import android.graphics.drawable.Drawable;

public class ResourceUtil {
    private static Context m_context;

    public static void SetContext(Context context) {
        m_context = context;
    }

    @SuppressLint("UseCompatLoadingForDrawables")
    public static Drawable GetDrawable(int resId){
        return m_context.getResources().getDrawable(resId);
    }

    @SuppressLint("UseCompatLoadingForDrawables")
    public static int GetColor(int resId){
        return m_context.getResources().getColor(resId);
    }

    public static String GetString(int resId) {
        return m_context.getResources().getString(resId);
    }
}

File: wallet-android/chapter7/code/app/src/main/java/com/digquant/util/ResourceUtil.kt
package com.digquant.util

import android.annotation.SuppressLint
import android.content.Context
import android.graphics.drawable.Drawable

object ResourceUtil {
    private var m_context: Context? = null
    fun SetContext(context: Context?) {
        m_context = context
    }

    @SuppressLint("UseCompatLoadingForDrawables")
    fun GetDrawable(resId: Int): Drawable {
        return m_context!!.resources.getDrawable(resId)
    }

    @SuppressLint("UseCompatLoadingForDrawables")
    fun GetColor(resId: Int): Int {
        return m_context!!.resources.getColor(resId)
    }

    fun GetString(resId: Int): String {
        return m_context!!.resources.getString(resId)
    }
}
